IllegalArgumentException: Comparison method violates its general contract!

If the last modified value of a file changes while TimSort is happening it can lead to an IllegalArgumentException:

```
Caused by: java.lang.IllegalArgumentException: Comparison method violates its general contract!
	at java.base/java.util.TimSort.mergeHi(TimSort.java:903)
	at java.base/java.util.TimSort.mergeAt(TimSort.java:520)
	at java.base/java.util.TimSort.mergeCollapse(TimSort.java:448)
	at java.base/java.util.TimSort.sort(TimSort.java:245)
	at java.base/java.util.Arrays.sort(Arrays.java:1515)
	at java.base/java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:353)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:503)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.eclipse.milo.opcua.stack.core.security.DefaultTrustListManager.pruneOldRejectedCertificates(DefaultTrustListManager.java:379)
	at org.eclipse.milo.opcua.stack.core.security.DefaultTrustListManager.addRejectedCertificate(DefaultTrustListManager.java:260)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.eclipse.milo.opcua.stack.core.security.DefaultCertificateValidator.verifyTrustChain(DefaultCertificateValidator.java:55)
	at org.eclipse.milo.opcua.stack.server.transport.uasc.UascServerAsymmetricHandler.onOpenSecureChannel(UascServerAsymmetricHandler.java:203)
	at org.eclipse.milo.opcua.stack.server.transport.uasc.UascServerAsymmetricHandler.decode(UascServerAsymmetricHandler.java:123)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:502)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:441)
	... 15 common frames omitted
```

--- a/opc-ua-stack/stack-core/src/main/java/org/eclipse/milo/opcua/stack/core/security/DefaultTrustListManager.java
+++ b/opc-ua-stack/stack-core/src/main/java/org/eclipse/milo/opcua/stack/core/security/DefaultTrustListManager.java
@@ -29,6 +29,7 @@ import java.security.cert.X509CRL;
 import java.security.cert.X509Certificate;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
@@ -372,10 +373,16 @@ public class DefaultTrustListManager implements TrustListManager, AutoCloseable 
         if (files != null && files.length >= MAX_REJECTED_CERTIFICATES) {
             int excessCount = files.length - MAX_REJECTED_CERTIFICATES;
 
+            // If last modified of any file changes during the sort it can lead
+            // to "IllegalArgumentException: Comparison method violates its general contract!"
+            // thrown from Java's TimSort implementation.
+            Map<File, Long> stableLastModified = new HashMap<>();
+            Arrays.stream(files).forEach(f -> stableLastModified.put(f, f.lastModified()));
+
             Arrays.stream(files)
                 .sorted(
-                    (o1, o2) ->
-                        (int) (o1.lastModified() - o2.lastModified()))
+                    (f1, f2) ->
+                        (int) (stableLastModified.get(f1) - stableLastModified.get(f2)))
                 .limit(excessCount + 1)
                 .forEach(file -> {
                     if (!file.delete()) {

